src/types/aws/ap/registration.ts

```diff
@@ -227,19 +235,20 @@ export function isDoneOrgDetails(
 export function isDoneFSAInquiry(
   data: SavedRegistration
 ): data is DoneFSAInquiry {
-  //could be false
-  return !(
-    (data.Registration as FSAInquiry)
-      .AuthorizedToReceiveTaxDeductibleDonations == null
-  );
+  const { Registration: reg } = data as DoneFSAInquiry;
+  return reg.AuthorizedToReceiveTaxDeductibleDonations != null;
 }
 
 export function isDoneDocs(data: SavedRegistration): data is DoneDocs {
-  return !!(data.Registration as TDocumentation).Documentation;
+  const { Registration: reg } = data as DoneDocs;
+  return (
+    !!reg.Documentation && reg.AuthorizedToReceiveTaxDeductibleDonations != null
+  );
 }
 
 export function isDoneBanking(data: SavedRegistration): data is DoneBanking {
-  return !!(data.Registration as BankingDetails).BankStatementFile;
+  const { Registration: reg } = data as DoneBanking;
+  return !!reg.BankStatementFile && !!reg.Documentation;
```

Comment:
also added asserting 1-previous step as a criteria that `this` step is really done.